src/resources/Catalogs/ProductListingConfiguration.ts

```diff
@@ -6,7 +6,7 @@ import {ProductListingConfigurationModel} from './ProductListingConfigurationInt
 export default class ProductListingConfiguration extends Resource {
     static baseUrl = `/rest/organizations/${API.orgPlaceholder}/catalogs`;
 
-    list(catalogId: string, productListingId: string) {
+    list(catalogId: string, productListingId: string): Promise<Record<string, ProductListingConfigurationModel[]>> {
```

Comment:
If I check the [swagger](https://platform.cloud.coveo.com/docs?urls.primaryName=Catalog#/Product%20Listings/rest_organizations_paramId_catalogs_paramId_productlistings_get) and other paginated call such as https://github.com/coveo/platform-client/blob/33b4e90d0bf7d4d2f154ae3f3532304d629ef97f/src/resources/Catalogs/ProductListing.ts#L15-L20

I think you should do the same as the code I linked, but for your call ;)

PSA: Discussions about pagination is currently being held at https://github.com/coveo/api-guidelines/pull/6

Reply:
Thanks for the link and yes pagination makes complete sense here.